### ISSUE TITLE
Internal Server Error

### ISSUE THREAD

**Issue by Redni** [strapi/strapi#328]
**Node.js version**: 8.8.1
**npm version**: 5.5.1
**Strapi version**: 3.0.0-alpha.6.4
**Operating system**: Arch Linux

`{"message":"An internal server error occurred","statusCode":500,"error":"Internal Server Error"}`

Example All: 
http://localhost:1337/letter

```
ERROR (5195 on inder): Cannot read property 'fetchAll' of undefined
    TypeError: Cannot read property 'fetchAll' of undefined
    at find (/home/inder/strapi/api/letter/controllers/Letter.js:18:51)
    at dispatch (/usr/lib/node_modules/strapi/node_modules/koa-router/node_modules/koa-compose/index.js:44:32)
    at next (/usr/lib/node_modules/strapi/node_modules/koa-router/node_modules/koa-compose/index.js:45:18)
    at dispatch (/usr/lib/node_modules/strapi/node_modules/koa-compose/index.js:42:32)
    at next (/usr/lib/node_modules/strapi/node_modules/koa-compose/index.js:43:18)
    at /usr/lib/node_modules/strapi/lib/middlewares/router/utils/routerChecker.js:28:11
    at dispatch (/usr/lib/node_modules/strapi/node_modules/koa-compose/index.js:42:32)
    at /usr/lib/node_modules/strapi/node_modules/koa-compose/index.js:34:12
    at dispatch (/usr/lib/node_modules/strapi/node_modules/koa-router/node_modules/koa-compose/index.js:44:32)
    at next (/usr/lib/node_modules/strapi/node_modules/koa-router/node_modules/koa-compose/index.js:45:18)
    at validator (/usr/lib/node_modules/strapi/node_modules/koa-router-joi/joi-router.js:327:38)
    at dispatch (/usr/lib/node_modules/strapi/node_modules/koa-router/node_modules/koa-compose/index.js:44:32)
    at next (/usr/lib/node_modules/strapi/node_modules/koa-router/node_modules/koa-compose/index.js:45:18)
    at parsePayload (/usr/lib/node_modules/strapi/node_modules/koa-router-joi/joi-router.js:253:62)
    at dispatch (/usr/lib/node_modules/strapi/node_modules/koa-router/node_modules/koa-compose/index.js:44:32)
    at next (/usr/lib/node_modules/strapi/node_modules/koa-router/node_modules/koa-compose/index.js:45:18)
```

Example One: 
http://localhost:1337/letter/5a0032edcfd2ef118ac6b109

```
ERROR (5195 on inder): Cannot read property 'fetch' of undefined
    TypeError: Cannot read property 'fetch' of undefined
    at findOne (/home/inder/strapi/api/letter/controllers/Letter.js:31:51)
    at dispatch (/usr/lib/node_modules/strapi/node_modules/koa-router/node_modules/koa-compose/index.js:44:32)
    at next (/usr/lib/node_modules/strapi/node_modules/koa-router/node_modules/koa-compose/index.js:45:18)
    at dispatch (/usr/lib/node_modules/strapi/node_modules/koa-compose/index.js:42:32)
    at next (/usr/lib/node_modules/strapi/node_modules/koa-compose/index.js:43:18)
    at /usr/lib/node_modules/strapi/lib/middlewares/router/utils/routerChecker.js:28:11
    at dispatch (/usr/lib/node_modules/strapi/node_modules/koa-compose/index.js:42:32)
    at /usr/lib/node_modules/strapi/node_modules/koa-compose/index.js:34:12
    at dispatch (/usr/lib/node_modules/strapi/node_modules/koa-router/node_modules/koa-compose/index.js:44:32)
    at next (/usr/lib/node_modules/strapi/node_modules/koa-router/node_modules/koa-compose/index.js:45:18)
    at validator (/usr/lib/node_modules/strapi/node_modules/koa-router-joi/joi-router.js:327:38)
    at dispatch (/usr/lib/node_modules/strapi/node_modules/koa-router/node_modules/koa-compose/index.js:44:32)
    at next (/usr/lib/node_modules/strapi/node_modules/koa-router/node_modules/koa-compose/index.js:45:18)
    at parsePayload (/usr/lib/node_modules/strapi/node_modules/koa-router-joi/joi-router.js:253:62)
    at dispatch (/usr/lib/node_modules/strapi/node_modules/koa-router/node_modules/koa-compose/index.js:44:32)
    at next (/usr/lib/node_modules/strapi/node_modules/koa-router/node_modules/koa-compose/index.js:45:18)
```

**Comment by Aurelsicoko** [comment]
@Redni Can you show me the line 18 in the Letter.js controller? It should be 

```js 
const data = await strapi.services.letter.fetchAll(ctx.query);
```

So, do you have a Letter.js service in `./api/letter/services/Letter.js`?

**Comment by Redni** [comment]
@Aurelsicoko I discovered a problem. To tell the truth, the previous name was different, I changed it for these logs because I did not want to share a unique name so it was not used publicly. It turned out that I missed a significant problem.

When adding "New Content" in the admin panel, any type of name:
**strapi.api**
**test.api**

are automatically converted to:
**strapiApi**
**testApi**

then we have:

`./api/strapiapi/controllers/StrapiApi.js`
`./api/strapiapi/models/StrapiApi.js`
`./api/strapiapi/services/StrapiApi.js`

The appeals are as follows:
```js
const data = await strapi.services.strapiApi.fetchAll (ctx.query);
```

So to summarize. Names that only consist of lowercase work perfectly. These consisting of small and large do not work at all.

**Comment by lauriejim** [comment]
Okay thank you for this issue, generated code is not completely correct.
The global model have to be `Strapiapi` or `Testapi` instead of `StrapiApi`
And in strapi object it's `strapi.service.strapiapi` or `strapi.model.strapiapi` instead of `strapi.mode.strapiApi`.

**Comment by RicardoVaranda** [comment]
I'm also getting a similar error:

```
ERROR (41203 on IE-MACP-RIVA-01): 

  TypeError: Cannot read property 'status' of undefined
      at strapi.app.use (/usr/local/lib/node_modules/strapi/lib/middlewares/boom/index.js:42:30)
      at <anonymous>
```

When trying to simply create and save a new content type.
